acceptance/tests/agent/agent_parses_json_catalog.rb

```diff
@@ -0,0 +1,20 @@
+test_name "C99978: Agent parses a JSON catalog"
+tag 'risk:medium'
+
+require 'json'
+
+step "Refresh the catalog" do
+  on(agents, puppet("agent --test --server #{master}"))
+end
+
+step "Agent parses a JSON catalog" do
+  agents.each do |agent|
+    # Is it there and can we read it?
+    json_catalog = File.join(agent.puppet['client_datadir'], 'catalog',
+                             "#{agent.puppet['certname']}.json")
+    on(agent, "[[ -r #{json_catalog} ]]", {:acceptable_exit_codes => [0]})
+
+    # Can the agent parse it as JSON?
+    on(agent, puppet("catalog find --terminus json --server #{master} > /dev/null"))
```

Comment:
Is this enough to ensure that a JSON catalog was retrieved? I am fine if the answer is, yes. I am just wondering if we should validate that an expected catalog element is contained in the JSON payload.

For example
```
on(agent, puppet("catalog find --terminus json --server #{master}")) do |result|
  require 'json'
  json = JSON.parse(result.stdout)
  assert_equal(true, json.has_key? 'name')
  assert_equal(#{agent}, json['name'])
end
```